generators/client/index.js

```diff
@@ -271,6 +279,9 @@ module.exports = class extends BaseGenerator {
                 if (this.configOptions.hibernateCache) {
                     this.hibernateCache = this.configOptions.hibernateCache;
                 }
+                if (this.configOptions.cacheImplementation) {
+                    this.cacheImplementation = this.configOptions.cacheImplementation;
```

Comment:
I guess its much easier to just replace `hibernateCache` variable everywhere with `cacheProvider` in one shot rather having a 2 variable solution. There will be more files changed but thats fine, i'll review it